sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2216,21 +2216,25 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         numArgs: Int,
         u: UnresolvedFunction): Expression = {
       func match {
-        case owg: SupportsOrderingWithinGroup if u.isDistinct =>
+        case owg: InverseDistributionFunction if u.isDistinct =>
           throw QueryCompilationErrors.distinctInverseDistributionFunctionUnsupportedError(
             owg.prettyName)
-        case owg: SupportsOrderingWithinGroup
+        case owg: InverseDistributionFunction
           if !owg.orderingFilled && u.orderingWithinGroup.isEmpty =>
```

Comment:
It seems to be the only check that we should relax for `ListAgg`. Should we just add a new flag to `SupportsOrderingWithinGroup` to allow empty ordering, instead of adding a new trait `InverseDistributionFunction`?

Reply:
I made it this way because all exceptions here have the `INVALID_INVERSE_DISTRIBUTION_FUNCTION` prefix. I decided it's easier to decouple functions support `WITHIN GROUP (ORDER BY)` from the inverse distribution functions than to generalize the error classes.

Reply:
But all inverse distribution functions support `WITHIN GROUP (ORDER BY)`? I don't really get the key point here.

Reply:
> But all inverse distribution functions support WITHIN GROUP (ORDER BY)?

Yes, I just don't want to change the error messages.
For some abstract extender of the `SupportsOrderingWithinGroup` trait that always wants to have `WITHIN GROUP`, the error should not be `INVALID_INVERSE_DISTRIBUTION_FUNCTION.WITHIN_GROUP_MISSING` but something like `INVALID_WITHIN_GROUP_EXPRESSION.WITHIN_GROUP_MISSING`. So I saw 2 options and chose the first: 
1. Extract inverse distribution functions into special traits and save these narrow error classes.
2. Add a flag in `SupportsOrderingWithinGroup` and rename `INVALID_INVERSE_DISTRIBUTION_FUNCTION` -> `INVALID_WITHIN_GROUP_EXPRESSION`=> change errors for existing functions

Reply:
I'm in favor of the flag approach as it's easier to understand. e.g. we can add `def allowsDistinct`, `def allowsEmptyOrdering` in `SupportsOrderingWithinGroup`.

We need to improve the error message a little bit. We should say `inverse distribution function 'percentile_cont' does not support DISTINCT` which provides the function name, so that it indicates that other inverse distribution functions may still support DISTINCT.

Reply:
Removed `InverseDistributionFunction` and renamed `INVALID_INVERSE_DISTRIBUTION_FUNCTION` -> `INVALID_WITHIN_GROUP_EXPRESSION`. Refactored messages a little so it will be clearer that the error is related only to one specific function.